daml/Marketplace/Issuance/CFI.daml

```diff
@@ -0,0 +1,45 @@
+module Marketplace.Issuance.CFI where
+
+-- |Classification of Finanical Instruments, denoted using ISO 10962:
+-- https://en.wikipedia.org/wiki/ISO_10962
+newtype CFI = CFI { code : Text } deriving (Eq)
+
+-- TODO: Add CFI Category/Group types
+instance Show CFI where
+  show (CFI "EXXXXX") = "Equity"
+  show (CFI "TCXXXX") = "Currency"
+  show (CFI "FXXXXX") = "Future"
+  show (CFI "JXXXXX") = "Forward"
+  show (CFI "DBXXXX") = "Bond"
+  show (CFI "OCEXXX") = "European Call"
+  show (CFI "RWXXPX") = "Put Warrant"
+  show (CFI "ECXXXX") = "Convertible Share"
+  show (CFI "XXXXXX") = "Unknown"
+  show _              = "Other"
+
+equity : CFI
+equity = CFI "EXXXXX"
+
+currency : CFI
+currency = CFI "TCXXXX"
+
+future : CFI
+future = CFI "FXXXXX"
+
+forward : CFI
+forward = CFI "JXXXXX"
+
+bond : CFI
+bond = CFI "DBXXXX"
+
+europeanCall : CFI
+europeanCall = CFI "OCEXXX"
+
+putWarrant : CFI
+putWarrant = CFI "RWXXPX"
+
+convertibleShare : CFI
+convertibleShare = CFI "ECXXXX"
+
+unknown : CFI
+unknown = CFI "XXXXXX"
```

Comment:
Hmm, that's a great point that `XXXXXX` is not in the standard...it might be better then to use an empty string or some other indicator when you are declining to classify something, since using `MMMXXX` for something would indicate it is specifically _not_ something in the mostly-exhaustive standard. I think `MMMXXX` does work better for a few of the places I used `XXXXXX` though (as I couldn't find them on the list).

Reply:
How do you think cryptocurrencies would be classified, out of interest?

Reply:
It looks like there's some argument on the internet about that, general consensus does seem to be that ISO 10962 isn't up to date enough for digital assets. If you wanted to shoehorn it in, I'd probably do `TCMXXX` (Referential instruments, currency, other) for the ones actually purporting to be currencies.